src/main/scala/mesosphere/marathon/core/storage/store/impl/zk/ZkPersistenceStore.scala

```diff
@@ -135,7 +136,7 @@ class ZkPersistenceStore(
         await(client.children(path).asTry) match {
           case Success(Children(_, _, nodes)) =>
             nodes.map { path =>
-              OffsetDateTime.parse(path, ZkId.DateFormat)
+              OffsetDateTime.parse(path, DateTimeFormatter.ISO_OFFSET_DATE_TIME)
```

Comment:
I find it confusing to have a general `ZkId.DateFormat` and then use `DateTimeFormatter.ISO_OFFSET_DATE_TIME` here without even a comment. See my suggestion above.

I *assume* your reasoning is:
_if any version had been stored using Java 9 before, it was serialized with ISO_OFFSET_DATE_TIME and therefore will fail to be read with `Timestamp.formatter`. Since we consider `Timestamp.formatter` a subset of ISO_OFFSET_DATE_TIME (without nanoseconds), we simply parse al version strings using that one as it will read dates both with and without nanos._

I think something like this would be a decent comment to not let someone think _Oh we forgot to use ZkId.DateFormat here_.